mobile/native/src/channel_fee.rs

```diff
@@ -128,12 +141,40 @@ impl ChannelFeePaymentSubscriber {
             .expect("Mutex to not be poisoned") = None;
     }
 
-    fn get_funding_transaction(&self) -> Option<EsploraTransaction> {
+    fn get_funding_transaction(&self) -> Option<(ChannelId, EsploraTransaction)> {
         self.open_channel_tx
             .lock()
             .expect("Mutex to not be poisoned")
             .clone()
     }
+
+    async fn wait_for_outbound_capacity(&self, channel_id: ChannelId) -> Result<()> {
+        tokio::time::timeout(Duration::from_secs(5), async {
+            loop {
+                let channel_details = self
+                    .channel_manager
+                    .get_channel_details(&channel_id)
+                    .expect("Channel details to exist");
```

Comment:
🔧 This could panic if the channel is closed whilst we wait for this.

Reply:
that would be very unlucky, but I've changed it. good find :)